main/src/library/Map.flix

```diff
@@ -1,7 +1,266 @@
 namespace Map {
 
-    def empty[k, v]: Map[k, v] = ???
+    //
+    // ## Map Construction
+    //
 
-    def insert[k, v](k: k, v: v, m: Map[k, v]): Map[k, v] = ???
+    ///
+    /// Returns the empty map.
+    ///
+    def empty[k, v]: Map[k, v] = Map(Nil)
 
+    ///
+    /// Returns the singleton map where key `k` is mapped to value `v`.
+    ///
+    def singleton[k, v](k: k, v: v): Map[k, v] = Map((k, v) :: Nil)
+
+    //
+    // ## Basic Operations
+    //
+
+    ///
+    /// Returns `true` if and only if `m` is the empty map, i.e. `Map(Nil)`.
+    ///
+    def null[k, v](m: Map[k, v]): Bool =
+        let Map(xs) = m;
+            List.isEmpty(xs)
+
+    ///
+    /// Returns `Some(v)` if `k -> v` is in `m`.
+    /// Otherwise returns `None`.
+    ///
+    def get[k, v](k: k, m: Map[k, v]): Option[v] = match m with {
+        case Map(Nil) => None
+        case Map((a, b) :: rs) => if (a == k) Some(b) else get(k, Map(rs))
+    }
```

Comment:
Im totally fine with this solution, but I wanted to show you an alternative which uses your own List code:

```
let Map(xs) = m;
  List.findMap(match (key, v) -> if (key != k) None else Some(v), xs)
```